web/src/js/index.js

```diff
@@ -1,473 +1,479 @@
+    executionLogTable: [],
+    resourceIdPlaceholder: "",
+    resourceList: [],
+    selectedComment: "",
+    selectedExpiration: 0,
+    selectedOwner: "",
+    selectedResource: "",
+    selectedResourceId: "",
+    selectedService: "",
+    serviceList: [],
+    serviceSettings: [],
+    serviceSettingsFlat: [],
+    showApiKeyPopup: true,
+    showExecutionLogListLoadingGif: false,
+    showExecutionLogLoadingGif: false,
+    showExecutionLogPopup: false,
+    showHelpPopup: false,
+    showAllowlistDeletePopup: false,
+    showAllowlistLoadingGif: false,
+    showAllowlistPopup: false,
+    allowlist: [],
+    allowlistDataTables: "",
+    allowlistSearchTerm: "",
+  },
+  methods: {
+    // Allowlist
+    closeAllowlistDeletePopup: function () {
+      this.selectedResourceId = "";
+      this.showAllowlistDeletePopup = false;
+    },
+    closeAllowlistInsertPopup: function () {
+      this.resourceIdPlaceholder = "";
+      this.resourceList = [];
+      this.selectedComment = "";
+      this.selectedExpiration = 0;
+      this.selectedOwner = "";
+      this.selectedResource = "";
+      this.selectedResourceId = "";
+      this.selectedService = "";
+      this.showAllowlistPopup = false;
+    },
+    createAllowlistEntry: function () {
+      let formData = {
+        resource_id:
+          this.selectedService +
+          ":" +
+          this.selectedResource +
+          ":" +
+          this.selectedResourceId,
+        owner: this.selectedOwner,
+        comment: this.selectedComment,
+      };
+
+      sendApiRequest(convertJsonToGet(formData), "POST");
+    },
+    createAllowlistEntryFromExecutionLog: function (
+      service,
+      resource,
+      resourceId
+    ) {
+      this.selectedService = service.toLowerCase().replace(/ /g, "_");
+      this.selectedResource = resource.toLowerCase().replace(/ /g, "_");
+      this.selectedResourceId = resourceId;
+      this.updateResourceList(this.selectedService);
+      this.closeExecutionLogPopup();
+      this.openAllowlistInsertPopup();
+    },
+    deleteAllowlistEntry: function (resourceId) {
+      let formData = {
+        resource_id: resourceId,
+      };
+
+      sendApiRequest(convertJsonToGet(formData), "DELETE");
+    },
+    extendAllowlistEntry: function (rowId) {
+      let row = this.allowlist[rowId - 1];
+      let formData = {
+        resource_id: row.resource_id,
+        expiration: row.expiration,
+        owner: row.owner,
+        comment: row.comment,
+      };
+
+      sendApiRequest(convertJsonToGet(formData), "PUT");
+    },
+    updateResourceId: function (service, resource) {
+      this.resourceIdPlaceholder =
+        this.serviceSettings[service][resource]["id"];
+    },
+    updateResourceList: function (service) {
+      this.resourceList = Object.keys(this.serviceSettings[service]);
+
+      // auto select if only 1 option exists
+      if (this.resourceList.length === 1) {
+        this.selectedResource = this.resourceList[0];
```

Comment:
Codacy has a fix for the issue: [Delete `␍`](https://app.codacy.com/gh/servian/aws-auto-cleanup/pullRequest?prid=9792602)
```suggestion
        this.selectedResource = this.resourceList[0];
```